firefox-ios/firefox-ios-tests/Tests/StorageTests/RustAutofillTests.swift

```diff
@@ -35,229 +55,186 @@ class RustAutofillTests: XCTestCase {
-                    if let updated = success {
-                        XCTAssert(updated)
-                    }
-                    XCTAssertNil(err)
-                    expectationUpdateCard.fulfill()
-
-                    self.autofill.getCreditCard(id: creditCard!.guid) { updatedCardVal, err in
-                        XCTAssertNotNil(updatedCardVal)
-                        XCTAssertNil(err)
-                        XCTAssertEqual(updatedCardVal!.ccExpYear, updatedCreditCard.ccExpYear)
-                        expectationCheckUpdateCard.fulfill()
-                    }
-                }
-            }
-        }
+    func testListCreditCards() async throws {
+        let cards = try await listCreditCards()
+        _ = try await addCreditCard()
+        let updatedCards = try await listCreditCards()
 
-        waitForExpectations(timeout: 3, handler: nil)
+        XCTAssertEqual(cards.count, 0)
+        XCTAssertEqual(updatedCards.count, 1)
     }
 
-    func testDeleteCreditCard() {
-        let expectationAddCard = expectation(description: "completed add card")
-        let expectationGetCard = expectation(description: "completed getting card")
-        let expectationDeleteCard = expectation(description: "delete card")
-        let expectationCheckDeleteCard = expectation(description: "check that no card exist")
-
-        addCreditCard { creditCard, err in
-            XCTAssertNotNil(creditCard)
-            XCTAssertNil(err)
-            expectationAddCard.fulfill()
-
-            self.autofill.getCreditCard(id: creditCard!.guid) { card, error in
-                XCTAssertNotNil(card)
-                XCTAssertNil(err)
-                XCTAssertEqual(creditCard!.guid, card!.guid)
-                expectationGetCard.fulfill()
-
-                self.autofill.deleteCreditCard(id: card!.guid) { success, err in
-                    XCTAssert(success)
-                    XCTAssertNil(err)
-                    expectationDeleteCard.fulfill()
-
-                    self.autofill.getCreditCard(id: creditCard!.guid) { deletedCreditCard, error in
-                        XCTAssertNil(deletedCreditCard)
-                        XCTAssertNotNil(error)
-
-                        let expectedError =
-                        "NoSuchRecord(guid: \"\(creditCard!.guid)\")"
-                        XCTAssertEqual(expectedError, "\(error!)")
-                        expectationCheckDeleteCard.fulfill()
-                    }
-                }
-            }
-        }
+    func testListAllAddressesEmpty() async throws {
+        let addresses = try await listAllAddresses()
+
+        XCTAssertEqual(addresses.count, 0, "Addresses count should be 0 for an empty list")
+    }
+
+    func testUpdateCreditCard() async throws {
+        let creditCard = try await addCreditCard()
+        let card = try await getCreditCard(id: creditCard.guid)
+        let updatedCreditCard = UnencryptedCreditCardFields(ccName: creditCard.ccName,
```

Comment:
Nit: i'd call this creditCardVal so we compare it with updatedCreditCarVal, and also for this test instead of comparing that credit card value are the same after update, i'd change those value do different one so we test that actually the values has changed. We can just change the values for UnencryptedCreditCardFields to custom one and change the comparison

Reply:
It makes sense. I've applied these changes and rebased the branch.